src/InputParser.js

```diff
@@ -0,0 +1,81 @@
+import { CUSTOM_DELIMITER, ERROR, GROUPING_REGEX, NEED_ESCAPE_CHARS } from './constants.js';
+class InputParser {
+  #delimiterSet;
+
+  constructor() {
+    this.#delimiterSet = new Set([',', ':']);
+  }
+
+  parse(input) {
+    const { customDelimiters, expression } = InputParser.#getParts(input);
+    this.#addDelimiters(customDelimiters);
+    const numbers = this.#expressionToNumber(expression);
+
+    return numbers;
+  }
+
+  static #getParts(input) {
+    const { customDelimiters, expression } = input.match(GROUPING_REGEX).groups;
+    if (
+      !customDelimiters
+      && (input.startsWith(CUSTOM_DELIMITER.START)
+      || input.includes(CUSTOM_DELIMITER.END))
+    ) {
+      throw new Error(`${ERROR.HEADER} 커스텀 구분자의 형식이 바르지 않습니다.`);
```

Comment:
```suggestion
      throw new Error(`${ERROR.HEADER} ${ERROR. INCORRECT_CUSTOM_DELIMITER_FORMAT}`);
```
문자열로 작성하기보단 예찬님께서 작성해주신 상수로 정의하는게 휴면 에러도 없을 것 같고 일관성도 지킬 수 있을 것 같습니다👍

Reply:
상수처리를 나중에 하면서 다 도려냈다고 생각했는데 누락된 부분이 있었네요! 감사합니다 👍🏻